src/jdk.management/share/classes/jdk/management/HotSpotAOTCacheMXBean.java

```diff
@@ -0,0 +1,98 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.management;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.PlatformManagedObject;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+/**
+ * Management interface for the JDK's Ahead of Time (AOT) Cache.
+ *
+ * <p> The management interface is registered with the platform {@link MBeanServer
+ * MBeanServer}. The {@link ObjectName ObjectName} that uniquely identifies the management
+ * interface within the {@code MBeanServer} is: "jdk.management:type=HotSpotAOTCache".
+ *
+ * <p> Direct access to the MXBean interface can be obtained with
+ * {@link ManagementFactory#getPlatformMXBean(Class)}.
+ *
+ * {@code HotSpotAOTCacheMXBean} defines one operation at this time to
+ * end the AOT recording. More operations and/or properties may be added in a
+ * future release.
+ *
+ * @since 26
+ */
+public interface HotSpotAOTCacheMXBean extends PlatformManagedObject {
+  /**
+   * If an AOT recording is in progress, ends the recording. This operation completes
+   * after the AOT artifacts have been completely written.
+   *
+   * <p>The JVM will start recording AOT artifacts upon start-up if certain JVM options are
+   * given in the command-line. The recording will stop when the JVM exits, or when
+   * the {@code endRecording} method is called. Examples:
+   *
+   * <p> java -XX:AOTCacheOutput=app.aot ....
+   *
+   * <blockquote>
+   *    The JVM will record optimization information about the current application
+   *    that will be used to generate the AOT cache file app.aot. In a future execution of this application,
+   *    -XX:AOTCache=app.aot can be provided to improve the application's
+   *    start-up and warm-up performance.
+   * </blockquote>
+   *
+   * <p> java -XX:AOTMode=record -XX:AOTConfiguration=app.aotconfig ....
+   *
+   * <blockquote>
+   *    The JVM will record optimization information about the current application
```

Comment:
The JVM records optimization information for the current application in the AOT configuration file {@code app.aotconfig}. ...